pkgs/os-specific/linux/zfs/generic.nix

```diff
@@ -50,9 +51,33 @@ let
     # If you don't do this your ZFS builds will fail on any non-standard (e.g.
     # clang-built) kernels.
     stdenv' = if kernel == null then stdenv else kernel.stdenv;
+
+    isLTS = kernel.passthru.isLTS or false;
   in
 
-  stdenv'.mkDerivation {
+  lib.warnIf (!(buildKernel -> !isLTS -> allowNonLTS)) ''
+    You are using the ZFS kernel module in combination with a non-LTS kernel:
+
+    ${kernel.modDirVersion}
+
+    Because the development cycles of OpenZFS and the Linux kernel do not line
+    up, it is likely that any given kernel that is not an LTS release will go
+    EOL (and will therefore be removed from Nixpkgs) before an OpenZFS release
+    is available that supports the (at that time) latest non-LTS kernel.
+
+    When that happens, you will *no longer be able to update* and NixOS/Nixpkgs
+    maintainers will not be able to help you.
+
+    Please use an LTS kernel (e.g. `pkgs.linuxPackages`) instead if you wish to
+    continue to use ZFS in the future in a way that NixOS/Nixpkgs can support.
```

Comment:
```suggestion
    When that happens, you will *no longer be able to update* and NixOS/Nixpkgs maintainers will
    not be able to help you,  unless you downgrade to an LTS kernel (e.g. `pkgs.linuxPackages`).
    
    To avoid this situation it is recommended that you only use ZFS with an LTS kernel.
```